brushfire-core/src/main/scala/com/stripe/brushfire/TDigest.scala

```diff
@@ -0,0 +1,70 @@
+package com.stripe.brushfire
+
+import com.tdunning.math.stats.TDigest
+import com.twitter.algebird.{Group, Semigroup}
+
+private [this] case object TDigestSemigroup extends Semigroup[TDigest] {
+  override def plus(l: TDigest, r: TDigest): TDigest = {
+    val td = TDigest.createDigest(math.max(l.compression(), r.compression()))
+    td.add(l)
+    td.add(r)
+    td
+  }
+
+  override def sumOption(iter: TraversableOnce[TDigest]): Option[TDigest] = {
+    iter.foldLeft(None: Option[TDigest]) {
+      case (None, el) =>
+        val td = TDigest.createDigest(el.compression())
+        td.add(el)
+        Some(td)
+
+      case (f@Some(acc), el) =>
+        acc.add(el)
+        f
+    }
+  }
+}
+
+case class TDigestSplitter[T](k: Int = 25, compression: Double = 100.0) extends Splitter[Double, Map[T, Long]] {
+  override type S = Map[Map[T, Long], TDigest]
+
+  override def split(parent: Map[T, Long], stats: S): Iterable[Split[Double, Map[T, Long]]] = {
+    def splitDists(q: Double)(target: Map[T, Long], digest: TDigest): (Map[T, Long], Map[T, Long]) = {
+      // the cumulative density reported is sometimes outside of [0,1] so we need to bound it. otherwise
+      // the estimated target distribution will be far from realistic... leading the Evaluator to
+      // frequently pick a suboptimal split point
+      val leftDist = (digest.cdf(q).max(0.0).min(1.0) * digest.size().toDouble).toLong
+      val rightDist = digest.size - leftDist
+      val left = target.mapValues(_ => leftDist).filter(_._2 > 0)
+      val right = target.mapValues(_ => rightDist).filter(_._2 > 0)
+      (left, right)
+    }
+
+    implicit val tds = TDigestSemigroup
+    val splits = for {
+      merged <- Semigroup.sumOption(stats.valuesIterator).toSeq
+      i <- 1 to k
+      q = merged.quantile(i.toDouble / k.toDouble).max(0.0).min(merged.size().toDouble)
+      (leftDist, rightDist) <- Semigroup.sumOption(stats.map(Function.tupled(splitDists(q))))
+      if leftDist.nonEmpty || rightDist.nonEmpty
+    } yield {
+      BinarySplit(LessThan(q), leftDist, Group.minus(parent, rightDist))
```

Comment:
I've wondered about the `Group.minus` bit here. Many of the splitters (maybe all aside from `QTree`) do it, so I'm wondering if this is used by anyone or if this should also be done in `QTreeSplitter`?